researchstack-sdk/src/main/java/org/sagebionetworks/bridge/researchstack/task/tracked/TrackedDataObjectCollection.java

```diff
@@ -0,0 +1,325 @@
+        byte[] jsonBytes = trackedDataCollectionJson.getBytes();
+        fileAccess.writeData(context, fullFilePath, jsonBytes);
+
+        return true;
+    }
+
+    /**
+     * @param context Can be app or activity, used to load tracked data collection
+     * @return a new TrackedDataObjectCollection object with the current one's changes reverted
+     */
+    public TrackedDataObjectCollection reset(Context context, List<TrackedDataObject> trackedDataObjectList) {
+        return loadSavedCollection(context, filename, trackedDataObjectList);
+    }
+
+    /**
+     * Reset the changes without committing them.
+     * @param filename that will be used to save this collection if commitChanges is called during its execution
+     */
+    protected static TrackedDataObjectCollection newInstance(
+            String filename,
+            List<? extends TrackedDataObject> trackedDataObjectList)
+    {
+        TrackedDataObjectCollection collection = new TrackedDataObjectCollection();
+        collection.filename = filename;
+        collection.setTrackedItems(trackedDataObjectList);
+        return collection;
+    }
+
+    /**
+     * @param context   Can be app or activity, used to load tracked data collection
+     * @param filename  The filename of the saved collection, must be unique
+     * @param trackedDataObjectList The list of TrackedDataObject that this collection uses
+     * @return          a new instance of the collection if none exists,
+     */
+    public static @NonNull TrackedDataObjectCollection loadSavedCollection(
+            Context context,
+            String filename,
+            List<? extends TrackedDataObject> trackedDataObjectList)
+    {
+        FileAccess fileAccess = verifyFileAccess();
+
+        if (fileAccess == null) {
+            return newInstance(filename, trackedDataObjectList);
+        }
+
+        String fullFilePath = TRACKED_DATA_STORE_PATH + File.separator + filename;
+        byte[] jsonBytes = fileAccess.readData(context, fullFilePath);
+
+        // If the collection does not exists, create a blank one we can edit
+        TrackedDataObjectCollection collection;
+        if (jsonBytes == null) {
+            collection = newInstance(filename, trackedDataObjectList);
+        } else { // parse the collection from the saved json
+            String trackedDataCollectionJson = new String(jsonBytes);
+            Gson gson = createGson();
+            collection = gson.fromJson(trackedDataCollectionJson, TrackedDataObjectCollection.class);
+        }
+        collection.setTrackedItems(trackedDataObjectList);
+        return collection;
+    }
+
+    /**
+     * @return a FileAccess object, if one is available and valid, null otherwise
+     */
+    protected static FileAccess verifyFileAccess() {
+        StorageAccess storageAccess = StorageAccess.getInstance();
+        if (storageAccess == null) {
+            LogExt.e(TrackedDataObjectCollection.class, "StorageAccess is null");
```

Comment:
I'd prefer that Log4j is used instead of LogExt in the Bridge packages, so the format of logging and log levels are controlled the same way between android-sdk and researchstack-sdk. Thoughts?